tests/test_optimize.py

```diff
@@ -671,3 +671,65 @@ def test_update_stepsize(self):
         eta2 = 0.1
         opt.update_stepsize(eta2)
         assert opt._stepsize == eta2
+
+    @pytest.mark.parametrize(
+        "opt",
+        [
+            GradientDescentOptimizer(stepsize),
+            MomentumOptimizer(stepsize, momentum=gamma),
+            NesterovMomentumOptimizer(stepsize, momentum=gamma),
+            AdagradOptimizer(stepsize),
+            RMSPropOptimizer(stepsize, decay=gamma),
+            AdamOptimizer(stepsize, beta1=gamma, beta2=delta),
+            RotosolveOptimizer(),
+        ],
+    )
+    class Test_over_opts:
+        def test_kwargs(self, opt, tol):
+            """Test that the keywords get passed and alter the function"""
+
+            def kwargs_func(x, c=1.0):
+                return (x - c) ** 2
+
+            x_new_one = opt.step(kwargs_func, 1.0, c=1.0)
+            x_new_two = opt.step(kwargs_func, 1.0, c=2.0)
+
+            if getattr(opt, "reset", None):
+                opt.reset()
+
+            assert x_new_one != pytest.approx(x_new_two, abs=tol)
+
+        @pytest.mark.parametrize(
+            "func, args",
+            [
+                (lambda x, y: x * y, (1.0, 1.0)),
+                (lambda x, y: x[0] * y[0], (np.array([1.0]), np.array([1.0]))),
+            ],
+        )
+        def test_multi_args(self, opt, func, args, tol):
+            """Test multiple arguments to function"""
+            x_new, y_new = opt.step(func, *args)
```

Comment:
Spy could also be used here to ensure that the function was called with the args you expect